### DISCUSSION TITLE
Collateral feature

### DISCUSSION THREAD

**Discussion by rndquu** [pavlovcik/uad-ui#653] [Q&A] [answered]
A bunch of questions to the community regarding the collateral feature

1. Are these issues https://github.com/ubiquity/ubiquity-dollar/issues?q=is%3Aissue+is%3Aopen+collateral required for the production deployment?
2.  Is it a correct example of the collateral feature:
	a) User deposits some ERC20 stablecoin (USDC, USDT, etc...) to the ubiquity pool  
	b) On deposit that ERC20 stablecoin is swapped for LUSD and LUSD is deposited to the ubiquity pool
	c) User gets UAD in return for ERC20 stablecoin (which is converted to LUSD under the hood)
	d) Using some AMO strategies (https://github.com/ubiquity/ubiquity-dollar/issues/610) we are trying to earn yield on the deposited LUSD tokens
3. Why LUSD is selected as an underlying collateral token?
4. Do we plan to add support for other underlying collateral tokens (like RAI) ?

**Comment by 0x4007** [comment, marked as answer]
> 1. 

- #347 is research which is not essential for the production deployment. 
- #611 is required unless it can be upgraded/added later, but the intent was to force thinking through the AMO support, and include those capabilities in the collateral contracts. 
- #610 can be added later.

> 2. 

The first step is to allow users to directly mint/redeem LUSD only. Once that is possible, we can create the "ZAP" contract to abstract the LUSD away from the user, by automatically swapping the user's USDC/USDT/DAI and depositing or redeeming the LUSD. 

Does that answer all of your question?

> 3. 

There's many reasons why I believe that LUSD is a superior "store of value" but not exactly great as a stablecoin for transactions (it floats above peg.) 
- I think it's the most censorship resistant stable-ish asset, and it has the smallest attack surface area due to it being a static protocol (there is no governance on changes to the protocol, and no contract upgrades or modifications are even possible according to what I understand.)  
- Except for hacks (they have [surpassed $4B TVL in the past](https://dune.com/queries/217940/409193) and still never got hacked!) the only viable attack vector that I am aware of is if the US government forces Chainlink to kill their oracle. Fortunately, Liquity thought about this and included a fallback Tellor oracle for a graceful system wind down. 

I can articulate all the reasons some time in the future but unfortunately I feel unwell today. I hope those answers are sufficient for now.

> 4. 

I included sUSD support in the spec to force thinking through the system working with more than one type of asset, but I wanted to disable that until we need to scale. The liquidity is not as great on sUSD or LUSD when compared to even the likes of FRAX, but we're more concerned with using these assets as a "store of value" vs for transactions if that makes sense. I think that LUSD and sUSD are fantastic stores of value, considering their architectures/risks. 

RAI I'm not a total expert on - looking at the price chart it seems like it's near its bottom. I heard politically it isn't doing too well since the idea of liquid staking derivatives (Lido) came around, and how the protocol is using pure ETH. I guess their community is torn between forking and deploying with a liquid staking derivative as collateral.
